src/app/room/room.component.ts

```diff
@@ -5,22 +5,31 @@ import { ActivatedRoute, Router } from '@angular/router';
 import { Player } from './models/player';
 import constants from '../constants';
 import { PlayerCreationModalComponent } from './player-creation-modal/player-creation-modal.component';
+import { AngularFirestore, AngularFirestoreCollection } from '@angular/fire/compat/firestore';
+import { Observable, of } from 'rxjs';
 
 @Component({
   selector: 'room',
   templateUrl: './room.component.html',
   styleUrls: ['./room.component.scss']
 })
 export class RoomComponent implements OnInit {
-  player: Player | undefined;
+  player = {} as Player;
```

Comment:
I know that compiler is complaining about this! let's just be careful when we do checks on undefined since undefined =/= {}.

Reply:
that is true, but I don't check for that anywhere I think no? This seems more elegant to me than the alternative of `.?`